pkg/controller/controller.go

```diff
@@ -257,3 +260,112 @@ func initResource(clientset kubernetes.Interface) error {
 
 	return nil
 }
+
+//RunGC starts garbage collector
+func (c *Controller) RunGC(d time.Duration) {
+	timer := time.NewTimer(d)
+	defer timer.Stop()
+	for {
+		<-timer.C
+		err := c.garbageCollect()
+		if err != nil {
+			c.logger.Warningf("Failed to cleanup resources: %v", err)
+		}
+	}
+}
+
+func (c *Controller) garbageCollect() error {
+	functionList := spec.FunctionList{}
+	tprClient, err := utils.GetTPRClient()
```

Comment:
Code like this makes it super-hard to inject a fake `RESTClient` during unittests.  Put the client into the `Controller` struct (as type `rest.Interface`) and just refer to it here rather than constructing your own.  See existing `Controller.clientset` for a similar example.